### PR TITLE
ISPN-14130 Add the memory option in the data Distribution chart

### PR DESCRIPTION
**Pull request by dpanshug**
https://issues.redhat.com/browse/ISPN-14130

### CODE REVIEW

File: src/app/Caches/DataDistributionChart.tsx [hunk 1]
```diff
@@ -49,186 +49,248 @@ const DataDistributionChart = (props: {
       }, [tablePagination, dataDistribution])
 
     const onSelectStatsOptions = (event, selection, isPlaceholder) => {
-        if(selection === t('caches.cache-metrics.data-distribution-option-entries'))
-            setStatsOption(DataDistributionStatsOption.TotalEntries);
-        else if(selection === t('caches.cache-metrics.data-distribution-option-memory'))
-            setStatsOption(DataDistributionStatsOption.MemoryEntries);
-        setIsOpenStatsOptions(false);
+      if (
```
Comment (karesti) [inline src/app/Caches/DataDistributionChart.tsx:52]: code that does not change should not be changing format

**Review by karesti — CHANGES_REQUESTED** [review]
there is something with the formatter of the code that is not ok.
could you please run prettier? npm run format

File: src/app/Caches/DataDistributionChart.tsx [hunk 2]
```diff
@@ -49,186 +49,248 @@ const DataDistributionChart = (props: {
...
+    };
 
     const buildCardContent = () => {
-        if (loading && !dataDistribution) {
-            return (
-                <Spinner size={'lg'} />
-            );
-        }
+      if (loading && !dataDistribution) {
+        return <Spinner size={'lg'} />;
```
Comment (karesti) [inline src/app/Caches/DataDistributionChart.tsx:85]: the good practice is to englobe on ()

File: src/app/Caches/DataDistributionChart.tsx [hunk 3]
```diff
@@ -49,186 +49,248 @@ const DataDistributionChart = (props: {
...
+      }
 
-        if (error != '') {
-            return (
-                <TableErrorState error={error} />
-            )
-        }
+      if (error != '') {
+        return <TableErrorState error={error} />;
```
Comment (karesti) [inline src/app/Caches/DataDistributionChart.tsx:89]: same here

File: src/app/Caches/DataDistributionChart.tsx [hunk 4]
```diff
@@ -49,186 +49,248 @@ const DataDistributionChart = (props: {
...
-                maxDomain = dataDistribution.reduce((max, entry) => {
-                    return Math.max(max, entry.total_entries);
-                } , 1);
-            }
-            else{
-                maxDomain = dataDistribution.reduce((max, entry) => {
-                    return Math.max(max, entry.memory_entries);
-                } , 1);
-            }
+      // Find max domain
```
Comment (karesti) [inline src/app/Caches/DataDistributionChart.tsx:92]: is there any change in this code ?

**Review by karesti — CHANGES_REQUESTED** [review]

**Comment by dpanshug** [comment]
@karesti code might be changing because of prettier formatting, will create another PR for improving it in other files also.

**Review by karesti — CHANGES_REQUESTED** [review]

**Comment by karesti** [comment]
@dpanshug needs rebase

**Comment by karesti** [comment]
@dpanshug did you do any changes on this PR ?

**Comment by dpanshug** [comment]
> @dpanshug did you do any changes on this PR ?

I've pushed the changes now.

**Review by karesti — CHANGES_REQUESTED** [review]
I'm testing the feature where the endpoint answers
``` json
[
  {
    "node_name": "3803a7d55fbe-10729",
    "node_addresses": [
      "172.17.0.2:7800"
    ],
    "memory_entries": 1,
    "total_entries": 1,
    "memory_used": 104
  }
]
```
And nothing is displayed. I added one entry to a cache with this config:

Cache config
```json
{
  "vvbvbvvbb": {
    "distributed-cache": {
      "owners": "1",
      "mode": "SYNC",
      "statistics": true,
      "encoding": {
        "key": {
          "media-type": "application/x-protostream"
        },
        "value": {
          "media-type": "application/x-protostream"
        }
      },
      "locking": {
        "isolation": "REPEATABLE_READ"
      },
      "memory": {
        "max-size": "56MB",
        "when-full": "REMOVE"
      }
    }
  }
}
```

**Comment by dpanshug** [comment]
@karesti I've updated the validation for displaying memory_used, now `memory_used` will work only when `max-size` is configured.

**Review by karesti — CHANGES_REQUESTED** [review]
rebase please!

**Review by karesti — APPROVED** [review]

**Pull request merged**